packages/expo-updates/ios/EXUpdates/DevLauncherAppController.swift

```diff
@@ -318,19 +318,19 @@ public final class DevLauncherAppController: NSObject, InternalAppControllerInte
   }
 
   public func checkForUpdate(success successBlockArg: @escaping (CheckForUpdateResult) -> Void, error errorBlockArg: @escaping (ExpoModulesCore.Exception) -> Void) {
-    errorBlockArg(NotAvailableInDevClientException())
+    errorBlockArg(NotAvailableInDevClientException("check for update"))
```

Comment:
should we use the js method names here, so people know exactly what to search their code for? so, `checkForUpdateAsync` in this case

Reply:
I agree with this, we should use the public API names

Reply:
Potentially! I'm curious how we'd phrase it. Something like: "checkForUpdateAsync cannot be called from a development client. A non-development build should be used to test this functionality." ?

Reply:
It's not that it can't be called, it's more that while it can be called it will always throw. So maybe "checkForUpdateAsync is not available in development client builds. A non-development build should be used to test this functionality."

Reply:
maybe:

```
Updates.checkForUpdateAsync() cannot be called from a development build. A release build should be used to test this functionality.
```

Reply:
The "release build" is technically incorrect as well since a debug build can also be used if the correct EX_UPDATE_NATIVE_DEBUG flags are set as well. The set of builds that need to be used are "another type of build build except one that uses dev client".

Reply:
maybe we can remove the second sentence then?

> Updates.checkForUpdateAsync() was called, but it is a no-op in development builds.

Reply:
It isn't a no-op, it will throw an exception...

Reply:
I like the idea of making it a single sentence that implies that the inverse of a development build is where it is supported. I went with:

```
Updates.checkForUpdateAsync() is not supported in development builds.
```